pkg/apis/networking/types.go

```diff
@@ -544,3 +544,73 @@ type ServiceBackendPort struct {
 	// +optional
 	Number int32
 }
+
+// CIDRConfig defines the CIDR and Mask size per IP Family(IPv4/IPv6).
+type CIDRConfig struct {
+	// Nodes may only have 1 range from each family.
+	// An IP block in CIDR notation ("10.0.0.0/8", "fd12:3456:789a:1::/64").
+	CIDR string
+
+	// PerNodeMaskSize is the mask size for node cidr.
+	// IPv4/IPv6 Netmask size (e.g. 25 -> "/25" or 112 -> "/112") to allocate to a node.
+	// Users would have to ensure that the kubelet doesn't try to schedule
+	// more pods than are supported by the node's netmask (i.e. the kubelet's
+	// --max-pods flag).
+	PerNodeMaskSize int32
+}
+
+// ClusterCIDRConfigSpec defines the desired state of ClusterCIDRConfig.
+type ClusterCIDRConfigSpec struct {
+	// NodeSelector defines which nodes the config is applicable to.
+	// An empty or nil NodeSelector functions as a default that applies to all nodes.
+	// +optional
+	NodeSelector *metav1.LabelSelector
```

Comment:
we were recommended to use v1.NodeSelector instead

Reply:
Updated the type for NodeSelector: See commit [Change ClusterCIDRConfig.NodeSelector type](https://github.com/kubernetes/kubernetes/pull/108290/commits/b1377880e83e12e6a998fe939091acf53dffa501)